ui/app/services/csp-event.js

```diff
@@ -0,0 +1,26 @@
+import Ember from 'ember';
+const { computed } = Ember;
+
+export default Ember.Service.extend({
+  init() {
+    this._super(...arguments);
+    this.handleCSP = Ember.run.bind(this, '_handleCSP');
+  },
+
+  events: [],
+
+  _handleCSP(event) {
+    this.get('events').addObject(event);
+  },
+
+  connectionViolations: computed.filterBy('events', 'violatedDirective', 'connect-src'),
+
+  attach() {
+    this.get('events').clear();
+    window.document.addEventListener('securitypolicyviolation', this.handleCSP, true);
+  },
+
+  remove() {
+    window.document.removeEventListener('securitypolicyviolation', this.handleCSP, true);
+  },
```

Comment:
Definitely not a change request, but I wondered how this service would look with ember-concurrency's `waitForEvent`. Untested:

```js
import Ember from 'ember';
import { task, waitForEvent } from 'ember-concurrency';

const { Service, computed } = Ember;

export default Service.extend({
  events: [],
  connectionViolations: computed.filterBy('events', 'violatedDirective', 'connect-src'),

  attach() {
    this.get('monitor').perform();
  },

  remove() {
    this.get('monitor').cancelAll();
  },

  monitor: task(function*() {
    this.get('events').clear();

    while (let event = waitForEvent(window.document.body, 'securitypolicyviolation')) {
      this.get('events').addObject(event);
    }
  }).restartable(),
});
```

Reply:
Oooo that is super nice, I may try updating that later. ❤ e-c.